controllers/openstackdataplanenode_controller.go

```diff
@@ -394,12 +394,37 @@ func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleNetworks(instance *dataplan
 	return instanceRole.Spec.NodeTemplate.Networks
 }
 
-// GetAnsibleVars returns a string value of ansible vars from the template unless it is set in the node
+// GetAnsibleVars returns a string value of ansible vars from merging the role template vars and node vars
 func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleVars(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) string {
-	if instance.Spec.Node.AnsibleVars != "" {
-		return instance.Spec.Node.AnsibleVars
+	// Merge the ansibleVars from the role into the value set on the node.
+	// Top level keys set on the node ansibleVars should override top level keys from the role AnsibleVars.
+	// However, there is no "deep" merge of values. Only top level keys are compared for merging
+
+	// Unmarshal the YAML strings into two maps
+	var role, node map[string]interface{}
+	roleYamlError := yaml.Unmarshal([]byte(instanceRole.Spec.NodeTemplate.AnsibleVars), &role)
+	if roleYamlError != nil {
+		r.Log.Error(roleYamlError, fmt.Sprintf("Failed to unmarshal YAML data from role AnsibleVars '%s'", instanceRole.Spec.NodeTemplate.AnsibleVars))
+		return ""
 	}
-	return instanceRole.Spec.NodeTemplate.AnsibleVars
+	nodeYamlError := yaml.Unmarshal([]byte(instance.Spec.Node.AnsibleVars), &node)
+	if nodeYamlError != nil {
+		r.Log.Error(nodeYamlError, fmt.Sprintf("Failed to unmarshal YAML data from node AnsibleVars '%s'", instance.Spec.Node.AnsibleVars))
+		return ""
+	}
+
+	// Merge the two maps
+	for k, v := range node {
+		role[k] = v
+	}
+
+	// Marshal the merged data back into a string of YAML
+	mergedYaml, err := yaml.Marshal(role)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Failed to unmarshal merged YAML data from role and node AnsibleVars '%s'", string(mergedYaml)))
+		return ""
+	}
+	return string(mergedYaml)
```

Comment:
we don't need to Marshal it,
we can remove the Unmarshal here:
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/controllers/openstackdataplanenode_controller.go#L292

Reply:
OK, that will be more efficient.
I was trying to not change the input and output types but the call on line 292 is the only one so it doesn't matter.

Reply:
OK, I've made this update. 

https://github.com/openstack-k8s-operators/dataplane-operator/compare/6c212c9acea21b15d87c71757cf2bc178e73ce6e..333630cf7f4779df3b35e4d2375fd072d458ec1d